pkg/network/ndmz/ndmz.go

```diff
@@ -226,35 +233,26 @@ func AttachNR(networkID string, nr *nr.NetResource) error {
 		return err
 	}
 
-	vethNR := "public"
-	vethDMZ := fmt.Sprintf("n-%s", nr.ID())
-
-	if !ifaceutil.Exists(vethDMZ, nil) || !ifaceutil.Exists(vethNR, nrNS) {
-		log.Debug().
-			Str("nr side", vethNR).
-			Str("dmz side", vethDMZ).
-			Msg("create veth pair to connect network resource and ndmz")
-
-		_ = ifaceutil.Delete(vethDMZ, nil)
-		_ = ifaceutil.Delete(vethNR, nrNS)
-
-		if _, _, err = ip.SetupVethWithName(vethDMZ, vethNR, 1500, nrNS); err != nil {
-			return errors.Wrap(err, "failed to create veth pair for to connect network resource and ndmz")
-		}
+	if _, err = macvlan.Create(publicIfaceName, BridgeNDMZ, nrNS); err != nil {
```

Comment:
How do you handle the case where the interface already exists ?

Reply:
This function is only called after creating a new network namespace and created the `public` interface inside the NS. So the only way to have an iface already exists here would be to call this function multiple times, which indicates a logic error on our side I think

Reply:
This function is only called after creating a new network namespace and created the public interface inside the NS. So the only way to have an iface already exists here would be to call this function multiple times, which indicates a logic error on our side I think.

Reply:
I don't think this is right.
A network resource can be already created when a new network reservation comes in. 
The current code deals with this gracefully and only update the wireguard configuration with the information of the new reservation.

Which mean a network resource can exists and this code be called anyone. All the methods related to network resource creation needs to be idempotent

Reply:
In this case I wonder if a `err != nil && !os.IsExists(err)` wouldn't be enough. But need to doulbe check what exactly happen in the `macvlan.Create` method

Reply:
I guess this is the code which would error:

https://github.com/threefoldtech/zos/blob/3b2f79066da4226e7df3bc5846656de599b00a91/pkg/network/macvlan/macvlan.go#L47-L49

which returns a custom error. So if we want to check for an os.IsExists error in the calling code, we also need to return that in the macvlan create code

Reply:
Actually no, this line will never return a exist error cause it is created random name and then only it is renamed from inside the network namespace.
So what you propose won't work.

I think we have to do the same as what the previous code was doing which is manually check if the interface already exists with `!ifaceutil.Exists(publicIfaceName, nrNS)`